tests/envs/llvm/gvn_sink_test.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""Tests for action space determinism."""
+import hashlib
+
+import pytest
+
+from compiler_gym.envs import LlvmEnv
+from tests.test_main import main
+
+pytest_plugins = ["tests.envs.llvm.fixtures"]
+
+
+@pytest.mark.xfail(strict=False)
+@pytest.mark.parametrize(
+    "benchmark_name",
+    [
+        "benchmark://cBench-v0/adpcm",
+        "benchmark://cBench-v0/bitcount",
+        "benchmark://cBench-v0/blowfish",
+        "benchmark://cBench-v0/bzip2",
+        "benchmark://cBench-v0/ghostscript",
+        "benchmark://cBench-v0/gsm",
+        "benchmark://cBench-v0/ispell",
+        "benchmark://cBench-v0/jpeg-c",
+        "benchmark://cBench-v0/jpeg-d",
+        "benchmark://cBench-v0/patricia",
+        "benchmark://cBench-v0/rijndael",
+        "benchmark://cBench-v0/stringsearch",
+        "benchmark://cBench-v0/stringsearch2",
+        "benchmark://cBench-v0/susan",
+        "benchmark://cBench-v0/tiff2bw",
+        "benchmark://cBench-v0/tiff2rgba",
+        "benchmark://cBench-v0/tiffdither",
+        "benchmark://cBench-v0/tiffmedian",
+    ],
+)
+def test_gvn_sink_non_determinism(env: LlvmEnv, benchmark_name: str):
+    """Regression test for -gvn-sink non-determinism.
+    See: https://github.com/facebookresearch/CompilerGym/issues/46
+    """
+    env.observation_space = "Ir"
+
+    checksums = set()
+    for i in range(1, 51):
```

Comment:
Ignorable comment:
I'm generally not a fan on odd looking constants. When I first see this line, I think "What's significant about 51?" I have to read through to find out the answer is nothing.
Consider passing num times as a param.

Reply:
Also consider doing some random steps first in case you just end up with no effect.
If doing rnd steps, consider having rnd seed param for repeatability.

Reply:
👍